src/servers/CreateServer.tsx

```diff
@@ -28,10 +27,11 @@ const CreateServer = (ImportServersBtn: FC<ImportServersBtnProps>, useStateFlagT
   const [ serversImported, setServersImported ] = useStateFlagTimeout(false, SHOW_IMPORT_MSG_TIME);
   const [ errorImporting, setErrorImporting ] = useStateFlagTimeout(false, SHOW_IMPORT_MSG_TIME);
   const handleSubmit = (serverData: ServerData) => {
-    const id = uuid();
+    const { newServers } = createServer({ ...serverData });
 
-    createServer({ ...serverData, id });
-    push(`/server/${id}`);
+    const newServer = Object.values(newServers)[0];
```

Comment:
This line is as ugly as it gets.

I didn't want to change the interface of `createServer` (mainly due to laziness, probably).

There should be: a) a better [-reading] way to get the only server that's been returned; and/or b) a better interface for the function.

Reply:
Well, you can't change the signature because it is an action creator, and the action getting dispatched is later passed to the reducer.

By changing the signature you would have to also adapt the reducer to support both use cases. So this is ok.